firmware/targets/furi_hal_include/furi_hal_i2c.h

```diff
@@ -40,34 +66,80 @@ void furi_hal_i2c_release(FuriHalI2cBusHandle* handle);
  * @param      address  I2C slave address
  * @param      data     pointer to data buffer
  * @param      size     size of data buffer
- * @param      timeout  timeout in ticks
+ * @param      timeout  timeout in milliseconds
  *
  * @return     true on successful transfer, false otherwise
  */
 bool furi_hal_i2c_tx(
     FuriHalI2cBusHandle* handle,
     const uint8_t address,
     const uint8_t* data,
-    const uint8_t size,
+    size_t size,
     uint32_t timeout);
 
+/** Perform I2C tx transfer, with additional settings.
+ *
+ * @param      handle   pointer to FuriHalI2cBusHandle instance
+ * @param      address  I2C slave address
+ * @param      ten_bit  whether the address is 10 bits wide
+ * @param      data     pointer to data buffer
+ * @param      size     size of data buffer
+ * @param      begin    how to begin the transaction
+ * @param      end      how to end the transaction
+ * @param      timeout  timeout in milliseconds
+ *
+ * @return     true on successful transfer, false otherwise
+ */
+bool furi_hal_i2c_tx_ext(
+    FuriHalI2cBusHandle* handle,
+    uint16_t address,
+    bool ten_bit,
+    uint8_t const* data,
+    size_t size,
+    I2CBegin begin,
+    I2CEnd end,
+    FuriHalCortexTimer timer);
```

Comment:
why do we need FuriHalCortexTimer here?

Reply:
To be able to have a common timer for multiple transactions.
For example `furi_hal_i2c_write_mem` uses the same timer both for sending `mem_addr` and the data to write.

Reply:
I've reverted this part, because: 
- For the sake of API simplicity
- CortexTimer is something that I'd rather to not to expose outside, it may be replaced in future
- Some further considerations needed: multithread OS may give time to another thread while we using I2C